src/win/i2c.cc

```diff
@@ -0,0 +1,318 @@
+    info.GetReturnValue().Set(cons->NewInstance(argc, argv));
+  }
+}
+
+NAN_METHOD(WinI2c::OpenSync) {
+  if (info.Length() < 1 ||
+      !info[0]->IsString()) {
+    return Nan::ThrowError(Nan::ErrnoException(EINVAL, "WinI2c::OpenSync",
+      "incorrect arguments passed to WinI2c::OpenSync"
+      "(string i2cControllerName)"));
+  }
+
+  Nan::HandleScope scope;
+  v8::String::Utf8Value str(info[0]->ToString());
+  std::string stdStr = std::string(*str);
+  std::wstring stdWStr;
+  stdWStr.assign(stdStr.begin(), stdStr.end());
+
+  String^ deviceSelector = I2cDevice::GetDeviceSelector(ref new Platform::String(stdWStr.c_str()));
+  DeviceInformationCollection^ i2cDeviceControllers = create_task(DeviceInformation::FindAllAsync(deviceSelector)).get();
+
+  if (nullptr == i2cDeviceControllers) {
+    Nan::ThrowError(Nan::ErrnoException(EPERM, "WinI2c::OpenSync", "DeviceInformation::FindAllAsync failed to return controller(s)"));
+  }
+
+  WinI2c* obj = Nan::ObjectWrap::Unwrap<WinI2c>(info.This());
+  obj->_i2cDeviceId = i2cDeviceControllers->GetAt(0)->Id;
+}
+
+NAN_METHOD(WinI2c::SetDevice) {
+  if (info.Length() < 1 ||
+      !info[0]->IsInt32()) {
+    return Nan::ThrowError(Nan::ErrnoException(EINVAL, "WinI2c::SetDevice",
+      "incorrect arguments passed to WinI2c::SetDevice"
+      "(int deviceAddress)"));
+  }
+
+  WinI2c* obj = Nan::ObjectWrap::Unwrap<WinI2c>(info.This());
+
+  if(nullptr == obj->_i2cDevice) {
+    I2cConnectionSettings^ settings = ref new I2cConnectionSettings(info[0]->Int32Value());
+    
+    //TODO: Expose as properties for the user to change
+    settings->BusSpeed = I2cBusSpeed::FastMode;
+    settings->SharingMode = I2cSharingMode::Exclusive;
+    
+    obj->_i2cDevice = create_task(I2cDevice::FromIdAsync(obj->_i2cDeviceId, settings)).get();
+    if (nullptr == obj->_i2cDevice) {
+      Nan::ThrowError(Nan::ErrnoException(EPERM, "WinI2c::SetDevice", "I2cDevice::FromIdAsync failed to return an i2c device"));
+    }
+  }
+}
+
+NAN_METHOD(WinI2c::CloseSync) {
+  WinI2c* obj = Nan::ObjectWrap::Unwrap<WinI2c>(info.This());
+
+  //From https://msdn.microsoft.com/en-us/library/windows.devices.i2c.i2cdevice.close.aspx
+  // You cannot call Close methods through Visual C++ component extensions(C++ / CX) on 
+  // Windows Runtime class instances where the class implemented IClosable.Instead, C++ / CX code 
+  // for runtime classes should call the destructor or set the last reference to null.
+  obj->_i2cDevice = nullptr;
+}
+
+NAN_METHOD(WinI2c::Read) {
+  Nan::ThrowError(Nan::ErrnoException(EPERM, "WinI2c::Read", "Not Implemented"));
+}
+
+NAN_METHOD(WinI2c::ReadPartial) {
+  if (info.Length() < 2 ||
+      !info[0]->IsInt32() ||
+      !info[1]->IsObject()) {
+    return Nan::ThrowError(Nan::ErrnoException(EINVAL, "WinI2c::ReadPartial",
+      "incorrect arguments passed to WinI2c::ReadPartial"
+      "(int length, Buffer buffer)"));
+  }
+
+  Nan::HandleScope scope;
```

Comment:
An implicit HandleScope is created on JavaScript accessible methods. Why is another one created here?